File: 1282-group-the-people-given-the-group-size-they-belong-to/1282-group-the-people-given-the-group-size-they-belong-to.swift
class Solution {
	func groupThePeople(_ groupSizes: [Int]) -> [[Int]] {
		var result: [[Int]] = []
		// Size, Array
		var map: [Int: [Int]] = [:]
		
		for i in 0 ..< groupSizes.count {
			let sz: Int = groupSizes[i]
			
			if map[sz] != nil {
				var arr: [Int] = map[sz]!
				arr.append(i)
				map[sz] = arr
				
				if arr.count == sz {
					result.append(arr)
					// clear
					map[sz] = []
				}
			}
			else {
				map[sz] = [i]
				
				if sz == map[sz]?.count {
					result.append(map[sz]!)
					map[sz] = []
				}
			}
		}
		
		print(map)
		return result
	}
}

File: 1282-group-the-people-given-the-group-size-they-belong-to/1282-group-the-people-given-the-group-size-they-belong-to.cs
public class Solution
{
    public IList<IList<int>> GroupThePeople(int[] groupSizes)
    {
        Dictionary<int, IList<int>> map = [];
        IList<IList<int>> result = [];

        for (int i = 0; i < groupSizes.Length; i++)
        {
            int groupSize = groupSizes[i];

            if (map.TryGetValue(groupSize, out IList<int>? value))
            {
                IList<int> list = value;
                list.Add(i);

                if (list.Count == groupSize)
                {
                    result.Add([.. list]);
                    map.Remove(groupSize);
                }
            }
            else
            {
                IList<int> l = [];
                l.Add(i);
                // case size = 1
                if (groupSize == 1)
                {
                    result.Add(l);
                    continue;
                }

                map[groupSize] = l;

            }
        }

        return result;
    }
}